File: src/main/java/com/project/bookstore/domain/book/Books.java
package com.project.bookstore.domain.book;

import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.Id;
import java.util.Date;

@Getter
@NoArgsConstructor
@Entity
public class Books {

    @Id
    private String ISBN;

    private String book_name;
    private String book_aut;
    private String book_trans;
    private Date book_date;
    private String book_cov;
    private int book_pri;
    private String book_gen;
    private String book_det;

    @Builder
    public Books(String ISBN, String book_name, String book_aut, String book_trans, Date book_date, String book_cov, int book_pri, String book_gen, String book_det) {
        this.ISBN = ISBN;
        this.book_name = book_name;
        this.book_aut = book_aut;
        this.book_trans = book_trans;
        this.book_date = book_date;
        this.book_cov = book_cov;
        this.book_pri = book_pri;
        this.book_gen = book_gen;
        this.book_det = book_det;
    }
}

File: src/main/resources/static/js/app/books.js
var main = {
    init : function() {
        var _this = this;

        $('#btn-save').on('click', function() {
            _this.save();
        });

        $('#btn-search').on('click', function () {
            _this.search();
        })
    },

    save : function() {
        var data = {
            isbn: $('#isbn').val(),
            bookName: $('#bookName').val(),
            bookAut: $('#bookAut').val(),
            bookTrans: $('#bookTrans').val(),
            bookDate: $('#bookDate').val(),
            bookHouse: $('#bookHouse').val(),
            bookCov: $('#bookCov').val(),
            bookPri: $('#bookPri').val(),
            bookGen: $('#bookGen').val(),
            bookDet: $('#bookDet').val()
        };

        $.ajax({
            type: 'POST',
            url: '/books/bookInsert',
            dataType: 'JSON',
            contentType:'application/json; charset=UTF-8',
            data: JSON.stringify(data)
        }).done(function() {
            alert("도서등록 완료");
            window.location.href = '/';
        }).fail(function(error) {
            console.log(error);
            alert(JSON.stringify(error));
        });
    },

    search : function() {
        var what = $("#searchWhat option:selected").val();

        console.log(what);

        var data = {
            searchWhat: what,
            searchKey: $('#searchKey').val()
        };
        
        if(what == "no") {
            alert("검색방법을 선택해주세요");
            return null;
        } else{
            window.location.href="/bookSearch?searchWhat="+data.searchWhat+"&searchKey="+data.searchKey;
        }

    }

};
main.init();